src/components/Account/Account.tsx

```diff
@@ -36,11 +44,13 @@ const AccountCard: FC<AccountProps> = ({ accountName, accountAddress, isShort =
     <>
       <Avatar size={24} src={DefaultAvatar} address={accountAddress} />
       <AccountInfoWrapper>
-        <Text>{accountName}</Text>
+        {!hideName && <Text>{accountName}</Text>}
         {!hideAddress && <AddressRow>
-          <Text size={'s'} color={'grey-500'}>
-            {isShort ? shortcutText(formatAddress(accountAddress) || '') : formatAddress(accountAddress) || ''}
-          </Text>
+          {hideName
```

Comment:
Split into smaller components. Use things like "isShort"/"hideName"/"hideAddress" as props.

Ternary in jsx are very hard to read and support especially when they are nested into one another